be/src/exprs/vectorized/encryption_functions.cpp

```diff
@@ -174,6 +175,34 @@ ColumnPtr EncryptionFunctions::md5sum(FunctionContext* ctx, const Columns& colum
     return result.build(ColumnHelper::is_all_const(columns));
 }
 
+ColumnPtr EncryptionFunctions::md5sum_numeric(FunctionContext* ctx, const Columns& columns) {
+    std::vector<ColumnViewer<TYPE_VARCHAR>> list;
+    list.reserve(columns.size());
+    for (const ColumnPtr& col : columns) {
+        list.emplace_back(ColumnViewer<TYPE_VARCHAR>(col));
+    }
+    auto size = columns[0]->size();
+    ColumnBuilder<TYPE_VARCHAR> result(size);
+    for (int row = 0; row < size; row++) {
+        Md5Digest digest;
+        for (auto& view : list) {
+            if (view.is_null(row)) {
+                continue;
+            }
+            auto v = view.value(row);
+            digest.update(v.data, v.size);
+        }
+        digest.digest();
+        StringParser::ParseResult parse_res;
+        uint128_t int_val =
+                StringParser::string_to_int<uint128_t>(digest.hex().c_str(), digest.hex().size(), 16, &parse_res);
+        DCHECK_EQ(parse_res, StringParser::PARSE_SUCCESS);
```

Comment:
what if parse error

Reply:
> what if parse error

this is impossible, the return result of Md5Digest is a hex number of up to 32 digits and uint128_t must not overflow.